src/view/dataElements/xdmObject.jsx

```diff
@@ -10,403 +10,252 @@ OF ANY KIND, either express or implied. See the License for the specific languag
+import useOnSchemaMetaSelectionChange from "./xdmObject/helpers/schemaSelection/useOnSchemaMetaSelectionChange";
+import ExtensionViewForm from "../components/extensionViewForm";
+import { reducer, ACTION_TYPES } from "./xdmObject/helpers/mainViewState";
 import "./xdmObject.styl";
-import InfoTipLayout from "../components/infoTipLayout";
-import debouncePromise from "../utils/debouncePromise";
+import loadDefaultSchema from "./xdmObject/helpers/schemaSelection/loadDefaultSchema";
+import getInitialFormStateUsingAsyncErrorReporting from "./xdmObject/helpers/schemaSelection/getInitialFormStateUsingAsyncErrorReporting";
 
-const STATUS_LOADING = "loading";
-const STATUS_LOADED = "loaded";
-const STATUS_ERROR = "error";
+const XdmObject = forwardRef(({ initInfo, resetForm }, ref) => {
+  const {
+    settings,
+    company: { orgId },
+    tokens: { imsAccess }
+  } = initInfo;
+  const reportAsyncError = useReportAsyncError();
+  const [state, dispatch] = useReducer(reducer, {
+    status: STATUS.INITIALIZING,
+    selectedNodeId: null,
+    defaultSelectedSandbox: null,
+    selectedSandbox: null,
+    defaultSelectedSchemaMeta: null,
+    selectedSchema: null,
+    showEditorNotReadyValidationError: false
+  });
 
-const SCHEMA_SEARCH_WAIT_MS = 250;
-const SCHEMA_SEARCH_DEFAULT = "";
+  const {
+    selectedNodeId,
+    status,
+    defaultSelectedSandbox,
+    selectedSandbox,
+    defaultSelectedSchemaMeta,
+    selectedSchema,
+    showEditorNotReadyValidationError
+  } = state;
+  const isEditorRenderable = status === STATUS.IDLE && Boolean(selectedSchema);
+  const onSandboxSelectionChange = useOnSandboxSelectionChange({
+    dispatch,
+    orgId,
+    imsAccess,
+    resetForm,
+    reportAsyncError
+  });
+  const onSchemaMetaSelectionChange = useOnSchemaMetaSelectionChange({
+    dispatch,
+    orgId,
+    imsAccess,
+    resetForm,
+    selectedSandbox,
+    reportAsyncError
+  });
 
-const XdmObject = ({
-  initInfo,
-  formikProps,
-  sandboxes,
-  selectedSandbox,
-  setSelectedSandbox,
-  schemasMeta,
-  schemasMetaStatus,
-  schemaOptionsSearch,
-  setSchemaOptionsSearch,
-  selectedSchemaMeta,
-  setSelectedSchemaMeta,
-  schemaStatus
-}) => {
-  const { values: formState } = formikProps;
-  const [selectedNodeId, setSelectedNodeId] = useState();
+  // It might seem desirable to take advantage of the useImperativeHandle hook here,
+  // which does something very similar. Unfortunately, when using the useImperative
+  // handle, and the user adds an item to an array field (or possibly other things),
+  // React sets ref.current to undefined until this component finishes its next render.
+  // During that same period of time, Formik attempts to validate the form (because
+  // it validates on change) and the parent components calls childRef.current.validateFormikState
+  // which throws an error because React has momentarily set childRef.current to undefined.
```

Comment:
This is weird, but I haven't really figured out a more appropriate way to deal with the situation.